IAM/Frequently_asked_questions.mediawiki

```diff
@@ -81,6 +81,32 @@ Two-factor authentication can be configured by using a Time-based One-Time Passw
    [[File:Github_-_2fa_success_message.png|400px]]
 10.  After 2FA has been enabled and you've saved your recovery codes, we recommend you sign out and back in to your account. In case of problems, such as a forgotten password or typo in your email address, you can use recovery codes to access your account and correct the problem.
 
+==== '''Q''': ''How can I set up two-factor authentication (2FA) for my github account, without using an app on my phone?'' ====
+
+If you would rather not install yet another app on your device, you can run the Firekey app (which is a 2-factor auth (TOTP) token generator for the Web) by browsing https://firekey.org/. <br>
+1.  In the upper-right corner of any github page, click your profile photo, then click Settings.
+   [[File:Github-settings.png|450px]]
+2.  In the user settings sidebar, click Security.
+   [[File:Github_-_security.png|150px]]
+3.  Under Two-factor authentication, click Set up two-factor authentication.
+   [[File:Github_-_set_up_2fa.png|400px]]
+4.  On the Two-factor authentication page, click Set up using an app.
+   [[File:Github_-_set_up_using_an_app.png|400px]]
+5.  To enable two-factor authentication you must save your two-factor recovery codes in a safe place, such as a password manager, by clicking Download, Print, or Copy. Your recovery codes will help get you back into your account if you've lost access. After saving your codes, click Next.
+   [[File:Github_-_recovery.png|450px]]
+6.  Get the secret key for your duo setup by either scanning the barcode (the URI generated should have the "secret" parameter) or by clicking the "enter this text code" link. Also, save this secret in a file. 
+   [[File:Github_-_enter_text_code.png|450px]]
+   [[File:Github_-_2fa_secret.png|300px]]
+7.  Copy the 2-factor secret.<br>
+8.  Navigate to https://firekey.org/ and add a new account. Enter a site name and paste the secret in "secret key" field, then click "Add".
```

Comment:
I don't think we can recommend this as this goes against our security fundamentals, effectively storing both factors on the same devices: https://infosec.mozilla.org/fundamentals/rationales.html#mfa

Reply:
To be explicit, the comparisons would be

# Comparisons

## User logs in on computer and uses TOTP code from app on phone

Not a problem, the second factor secret is not stored on the same device as the one being used to login (it's stored in the phone's sequestered storage for the app)

## User logs in on computer and uses TOTP code from firekey

Not a problem, the second factor secret is not stored on the same device as the one being used to login (it's stored in the phone's browser's localStorage)

## User logs in on phone and uses TOTP code from app on phone

The problem here is that the phone has access to both the password and the second factor secret. If the phone is compromised then the account is compromised.

## User logs in on phone and uses TOTP code from firekey

The problem here is that the phone has access to both the password and the second factor secret. If the phone is compromised then the account is compromised.

# Tangential issues

It's worth noting that the second factor secret is stored in different places with potentially different security profiles if you're talking about an app or a website

@gdestuynder can speak with more authority about where the duo or google authenticator apps store the secret and what the mobile OS security model (assuming it's not rooted) means for the security around that secret.

For Firekey, it's storing the secret in the browser's localStorage which has a different set of attacks.